pkg/client/openstack/project/client.go

```diff
@@ -214,3 +221,23 @@ func (c *projectClient) getFlavors() ([]models.Flavor, error) {
 
 	return result, err
 }
+
+func (c *projectClient) getAvailabilityZones() ([]models.AvailabilityZone, error) {
+	result := []models.AvailabilityZone{}
+	allPages, err := availabilityzones.List(c.ComputeClient).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	zones, err := availabilityzones.ExtractAvailabilityZones(allPages)
+	if err != nil {
+		return nil, err
+	}
+	for _, zone := range zones {
+		// only report on available zones
+		if zone.ZoneState.Available {
```

Comment:
I'm wondering why only report an available availability zones? The swagger type has a boolean flag to indicate if an AZ is available or not.

Reply:
Removed both the filter and the boolean from the swagger spec. My rational is that the AVZ will only ever be temporarily unavailable.

I cross-checked that we don't have workaround AVZ that are permanently unavailable. Looks good. So far.